src/jllvm/materialization/ByteCodeCompileLayer.cpp

```diff
@@ -955,8 +955,39 @@ void codeGenBody(llvm::Function* function, const Code& code, const ClassFile& cl
             }
             // TODO: FALoad
             // TODO: FAStore
-            // TODO: FCmpG
-            // TODO: FCmpL
+            case OpCodes::FCmpG:
+            case OpCodes::FCmpL:
+            {
+                llvm::Value* rhs = operandStack.pop_back(builder.getFloatTy());
+                llvm::Value* lhs = operandStack.pop_back(builder.getFloatTy());
+
+                llvm::Value* notEqual = builder.CreateFCmpUNE(lhs, rhs);
+                llvm::Value* otherCmp;
+                llvm::Value* defaultCase;
+
+                switch (opCode)
```

Comment:
I'll be honest, this is a lot shorter and nicer than I thought it'd be.
Having thought through it and as is also shown by the test I think its correct but also non-trivial to understand.

Could you maybe add some comments here about the logic? Things like why UNE was chosen (has to return 1 for NaNs), Why SExt and Zext, etc etc. Just your general thought process